apps/lib/apps.c

```diff
@@ -783,7 +783,7 @@ int load_key_certs_crls(const char *uri, int maybe_stdin,
         goto end;
 
     failed = NULL;
-    while (!OSSL_STORE_eof(ctx)) {
+    while (cnt_expectations > 0 && !OSSL_STORE_eof(ctx)) {
```

Comment:
This is not right for the case where just pparams is NON NULL..

You can see this a few lines up where it conditional sets expect = 0;

Reply:
e.g. this fails..
````
apps/openssl ecparam -out ecparam.tst -in ./tes
t/recipes/15-test_ecparam_data/valid/wap-wsg-idm-ecid-wtls7-explicit.pem
````

Reply:
Maybe this is the wrong way to do this and we need the bio to not return eof true until it is finished?

Reply:
No, IMO this is the right way and it just needed to properly set cnt_expectations and expect for the params.

Reply:
Not sure I understand..
If you have read to the end of the file isn't it finished at that point??

Reply:
The cnt_expectations > 0 is there to make the loop exit early. To not read until the end of the file actually.